ctapipe/image/tests/test_extractor.py

```diff
@@ -550,8 +558,9 @@ def test_global_peak_window_sum_with_pixel_fraction(subarray):
         apply_integration_correction=False,
     )
 
-    charge, peak_time = extractor(waveforms, tel_id, selected_gain_channel)
+    charge, peak_time, is_valid = extractor(waveforms, tel_id, selected_gain_channel)
```

Comment:
At this point, shouldn't the extractor just return the `CameraImageContainer`?

Reply:
yes, perhaps that is simpler now that the return value is so complicated.  In the end that's what we fill in.  However, doesn't LST have it's own subclass of this container? Or am I misremembering?

Reply:
It's a bit of a design decision: in general we had lower-level classes/functions return bare information so that they can be used without the need for a Container hierarchy, but here is maybe is simpler

Reply:
>  However, doesn't LST have it's own subclass of this container? Or am I misremembering?

No, "just" for the `ArrayEventContainer` that has an additional `.lst` field.

Reply:
https://github.com/cta-observatory/ctapipe_io_lst/blob/fd8ae42d2afadf2cbde4f80fa359a51822b9b321/ctapipe_io_lst/containers.py#L117

Reply:
> It's a bit of a design decision: in general we had lower-level classes/functions return bare information so that they can be used without the need for a Container hierarchy, but here is maybe is simpler

Here it's not a hierarchy, right? It's just like `hillas_parameters` and the other image parameters a leaf node in the container tree.

Reply:
Ok, sounds like it doesn't break LSTs, then, so I could just return the container.